### PR TITLE
Update Node, simplify Jenkinsfile

### PR DESCRIPTION
**Pull request by zbynek**
Node environment should not be production if we want dev dependencies to install.
The --tap flag for test stage was missing and all npm commands can go to a single stage for simplicity.

### CODE REVIEW

**Comment by zbynek** [comment]
@dduportal could we remove the restriction that PRs have to be approved in this repo?

**Comment by dduportal** [comment]
> @dduportal could we remove the restriction that PRs have to be approved in this repo?

Yes: what is the blocker for you (so we document it at least here for future selves): not enough maintainers available so it slow us down?

**Comment by dduportal** [comment]
Sorry, I clicked the wrong button: reopening and re-enabling automerge

File: Jenkinsfile
Comment (dduportal) [inline Jenkinsfile:33]: This NodeJS version is old as hell 😱 Not blocking the PR, but removing the agent block and using any kind of agent (either container `maven-21` or VM with `docker`)  would provide modern NodeJS LTS (20.18.x right now).

File: Jenkinsfile
Comment (dduportal) [inline Jenkinsfile:38]:
```suggestion
        sh 'npm clean-install'
```

Using `npm install` is considered a bad practice as it will try to update non locked dependencies, while clean-install reads from the lock file.

Not a blocker, but better practice (and additional safety nets when using old versions like here)

**Review by dduportal — APPROVED** [review]
A few non blocking suggestions

**Pull request merged**
